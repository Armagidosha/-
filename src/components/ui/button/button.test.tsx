import { Button } from "./button";
import renderer from "react-test-renderer"
import { render, fireEvent, screen } from "@testing-library/react";

describe("button component", () => {
  
  test("button without text", () => {
    const button = renderer
    .create(<Button />)
    .toJSON();
    expect(button).toMatchSnapshot();
  })

test("button with text", () => {
  const button = renderer
    .create(<Button text="Test Button" />)
    .toJSON();
    expect(button).toMatchSnapshot();
})

test("button is disabled", () => {
  const button = renderer
    .create(<Button disabled />)
    .toJSON();
    expect(button).toMatchSnapshot();
})

test("button is loading", () => {
  const button = renderer
    .create(<Button isLoader={true} />)
    .toJSON();
    expect(button).toMatchSnapshot();
})

test("button with callBack", () => {
  window.alert = jest.fn();

  render(<Button text="Testing Button" onClick={() => alert('Button Tested')}/>)
  const button = screen.getByText('Testing Button')
  fireEvent.click(button)
  expect(window.alert).toHaveBeenCalledWith('Button Tested')
})
})