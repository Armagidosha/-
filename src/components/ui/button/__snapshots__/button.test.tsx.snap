// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component button is loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button component button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test Button
  </p>
</button>
`;

exports[`button component button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
